internal/pkg/provision/request.go

```diff
@@ -30,10 +30,10 @@ type NetworkRequest struct {
 }
 
 // NodeRequests is a list of NodeRequest.
-type NodeRequests []NodeRequest
+type NodeRequests []*NodeRequest
```

Comment:
is there a strong case for using pointer here?

Go is really efficient at copying data structures around, and I guess it might skip copying if it can prove it's safe.

Having copy of the data eliminates lots of small cases here and there. 

Of course there's a probably a lot of cases when having a pointer makes sense (like Mutex being a field), but is there anything like this here?

Reply:
I was just having trouble tweaking the configdata value in the node requests without it being a pointer. Let me take a second look.